src/training_hub/algorithms/speculative_decoding.py

```
@@ -0,0 +1,1173 @@
+
+        cmd.extend([
+            "--preprocessed-data", data_output_dir,
+            "--endpoint", vllm_endpoint,
+            "--output", hidden_states_path,
+            "--concurrency", str(concurrency),
+        ])
+        if max_samples:
+            cmd.extend(["--max-samples", str(max_samples)])
+
+        logger.info("Generating hidden states: %s", " ".join(cmd))
+        result = subprocess.run(cmd, capture_output=True, text=True, timeout=1800)
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)